### PR TITLE
feat: Add option Check On Testcases With Empty Output

### LINKED ISSUE #208: Checker return no verdict when Expected Output is empty
**Screenshots**

### PR DESCRIPTION
**Pull request by tb148**
Add an option to check answers on testcases with empty outputs.

## Related Issue
Fix #208.

## Type of changes

- [x] Bug fix (changes which fix an issue)
- [ ] New feature (changes which add functionality)
- [ ] Documentation (changes which modify the documentation only)
- [ ] Style change (changes which do not affect the meaning of the code: code formatting, etc.)
- [ ] Refactor (changes which affect the meaning of the code but neither fix a bug nor add a feature)
- [ ] Performance improve (changes which improve performance)
- [ ] Test (changes which add tests)
- [ ] Build (Changes that affect the build system or external dependencies)
- [ ] CI (changes to CI configuration files and scripts)
- [ ] Chore (changes which do not belong to any type above)
- [ ] Revert (revert previous changes)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/cpeditor/cpeditor/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested these changes locally, and this fixes the bug/the new feature behaves as the expectation.
- [ ] The settings file in the old version can be used in the new version after this change.
- [x] These changes only fix a single bug/introduces a single feature. (Otherwise, open multiple Pull Requests instead, unless these bugs/features are closely related.)
- [ ] The commit messages are clear and detailed. (Otherwise, use `git reset` and commit again, or use `git rebase -i` and `git commit --amend` to modify the commit messages.)
- [x] These changes don't remove an existing feature. (Otherwise, add an option to disable this feature instead, unless it's necessary to remove this feature.)
- [ ] If there are changes of the text displayed in the UI, I have wrapped them in `tr()` or `QCoreApplication::translate()`.
- [x] I have documented these changes in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/doc/CHANGELOG.md), or these changes are not notable.

### CODE REVIEW

**Comment by ouuan** [comment]
You can enable GitHub Actions in [your repository](https://github.com/tb148/cpeditor/actions) so that your codes can be formatted automatically.

However, it's better to format locally, and you can enable "format on save" in your IDE (if it's supported).

**Comment by coder3101** [comment]
Or by running this [script](https://github.com/cpeditor/cpeditor/blob/master/tools/format.sh) on bash. If you may need to install clang-format. (LLVM on Windows)

**Comment by tb148** [comment]
The translation of this option isn't created yet. You can ask me to add it if you want.

**Comment by coder3101** [comment]
Required Steps:

- Complete checklist of the PR (Contribution.md doc must be agreed to)

- Add your change to changelog file in `doc/CHANGELOG.md` under **Unreleased**, under **Added** subsection.


Optional Steps:
- If you can you should add translation for Chinese language.

File: src/Settings/settings.json [hunk 1]
```diff
@@ -763,5 +763,11 @@
         "desc": "Run your codes on empty test cases",
         "type": "bool",
         "tip": "Run your code on all non-hidden test cases even if the input is empty."
+    },
+    {
+        "name": "Check On Testcases With Empty Output",
+        "desc": "Check your answer on test cases with empty output",
+        "type": "bool",
+        "tip": "Check your answer on all non-hidden test cases even if the output is empty."
```
Comment (ouuan) [inline src/Settings/settings.json:771]:
```suggestion
        "tip": "Check your answer even if your output or the expected output is empty."
```

File: translations/zh_CN.ts [hunk 2]
```diff
@@ -2147,6 +2152,16 @@ from Competitive Companion again, the old file will be opened.</source>
         <source>Run your code on all non-hidden test cases even if the input is empty.</source>
         <translation>在所有未隐藏的测试点运行你的代码，即使输入为空。</translation>
     </message>
+	<message>
+        <location filename="../build/generated/SettingsInfo.cpp" line="131"/>
+        <source>Check your answer on test cases with empty output</source>
+        <translation>在标准输出为空的测试点上评测你的代码</translation>
```
Comment (ouuan) [inline translations/zh_CN.ts:2158]:
```suggestion
        <translation>在输出为空的测试点上检查输出的正确性</translation>
```

File: translations/zh_CN.ts [hunk 3]
```diff
@@ -2147,6 +2152,16 @@ from Competitive Companion again, the old file will be opened.</source>
...
+    </message>
+    <message>
+        <location filename="../build/generated/SettingsInfo.cpp" line="131"/>
+        <source>Check your answer even if your output or the expected the output is empty.</source>
+        <translation>在所有测试点评测你的代码，即使输出为空。</translation>
```
Comment (ouuan) [inline translations/zh_CN.ts:2163]:
```suggestion
        <translation>即使你的输出或答案的输出为空，依然检查输出的正确性。</translation>
```

**Comment by tb148** [comment]
> 
> 
> Did you make this change manually without running `tools/updateTranslation.sh`?

Yes, because somehow it doesn't works on my computer (Windows 10)

**Comment by ouuan** [comment]
> Yes, because somehow it doesn't works on my computer (Windows 10)

Maybe you can run it by git bash (haven't tested yet).

**Comment by ouuan** [comment]
@allcontributors add @tb148 as a contributor for codes.

**Pull request merged**